e2e failure to get unpriv user ids (new sudo versions)

### Version of Singularity:

`master`

### Actual behavior

On Fedora 33 with sudo `sudo-1.9.5p2-1.fc33.x86_64` we need to walk up the process chain until *both* `uid !=0` and `gid !=0` when finding our unpriv account for the e2e tests. Changes in sudo mean that at present we can land on a process that doesn't satisfy this at present.

```
01:43 PM $ make -C builddir e2e-test
make: Entering directory '/home/dave/Git_hpcng/singularity/builddir'
 TEST sudo go test [e2e]
rm -f /home/dave/Git_hpcng/singularity/builddir/e2e-cmd-coverage
cd /home/dave/Git_hpcng/singularity && \
	SINGULARITY_E2E_COVERAGE=/home/dave/Git_hpcng/singularity/builddir/e2e-cmd-coverage \
	scripts/e2e-test -v
failed to retrieve user information
FAIL	github.com/sylabs/singularity/e2e	0.043s
FAIL
make: *** [Makefile:560: e2e-test] Error 1
make: Leaving directory '/home/dave/Git_hpcng/singularity/builddir'
```

Diff fixes the issue:

```diff
diff --git a/e2e/internal/e2e/init/init_linux.go b/e2e/internal/e2e/init/init_linux.go
index b2b567e41..0ab738bc5 100644
--- a/e2e/internal/e2e/init/init_linux.go
+++ b/e2e/internal/e2e/init/init_linux.go
@@ -68,7 +68,7 @@ static int getUnprivIDs(pid_t pid, uid_t *uid, gid_t *gid) {
                return -1;
        }
        pid_t ppid = getProcInfo(pid, uid, gid);
-       if ( *uid == 0 ) {
+       if ( *uid == 0 || *gid == 0 ) {
                return getUnprivIDs(ppid, uid, gid);
        }
        return 0;
```

--- a/e2e/internal/e2e/init/init_linux.go
+++ b/e2e/internal/e2e/init/init_linux.go
@@ -68,7 +68,7 @@ static int getUnprivIDs(pid_t pid, uid_t *uid, gid_t *gid) {
 		return -1;
 	}
 	pid_t ppid = getProcInfo(pid, uid, gid);
-	if ( *uid == 0 ) {
+	if ( *uid == 0 || *gid == 0 ) {
 		return getUnprivIDs(ppid, uid, gid);
 	}
 	return 0;

